### PR TITLE
Add example configuration for datasource `AliYun`

### PR DESCRIPTION
**Pull request by bittopaz**
## Proposed Commit Message

```
Add example configuration for datasource `AliYun`
```

## Checklist:

 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/hacking.html)
 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any documentation accordingly

### CODE REVIEW

File: doc/rtd/topics/datasources/aliyun.rst [hunk 1]
```diff
@@ -12,6 +12,22 @@ The Alibaba Cloud metadata service is available at the well known url
 Alibaba Cloud ECS on `metadata
 <https://www.alibabacloud.com/help/zh/faq-detail/49122.htm>`__.
 
+Configuration
+-------------
+The following configuration can be set for the datasource in system
+configuration (in `/etc/cloud/cloud.cfg` or `/etc/cloud/cloud.cfg.d/`).
```
Comment (OddBloke) [inline doc/rtd/topics/datasources/aliyun.rst:18]: This is reStructuredText, not Markdown, so you need double ticks here:

```suggestion
configuration (in ``/etc/cloud/cloud.cfg`` or ``/etc/cloud/cloud.cfg.d/``).
```

File: doc/rtd/topics/datasources/aliyun.rst [hunk 2]
```diff
@@ -12,6 +12,22 @@ The Alibaba Cloud metadata service is available at the well known url
...
+
+An example configuration with the default values is provided below:
+
+.. sourcecode:: yaml
+
+  datasource:
+    AliYun:
+      metadata_urls: ["http://100.100.100.200", "http://instance-data:8773"]
```
Comment (OddBloke) [inline doc/rtd/topics/datasources/aliyun.rst:26]: This doesn't match the AliYun default: https://github.com/canonical/cloud-init/blob/e5c67478136e1942a981ee18785ceb26579d0041/cloudinit/sources/DataSourceAliYun.py#L13

File: doc/rtd/topics/datasources/aliyun.rst [hunk 3]
```diff
@@ -12,6 +12,22 @@ The Alibaba Cloud metadata service is available at the well known url
...
+      support_xen: false
+      timeout: 5
+      max_wait: 300
```
Comment (OddBloke) [inline doc/rtd/topics/datasources/aliyun.rst:29]: These don't match the AliYun defaults (which are inherited from the EC2 datasource): https://github.com/canonical/cloud-init/blob/e5c67478136e1942a981ee18785ceb26579d0041/cloudinit/sources/DataSourceEc2.py#L69-L70

**Review by OddBloke — CHANGES_REQUESTED** [review]
Thanks for the submission, @bittopaz! I've left a few comments inline.

File: doc/rtd/topics/datasources/aliyun.rst [hunk 4]
```diff
@@ -12,6 +12,22 @@ The Alibaba Cloud metadata service is available at the well known url
...
+configuration (in `/etc/cloud/cloud.cfg` or `/etc/cloud/cloud.cfg.d/`).
+
+An example configuration with the default values is provided below:
+
+.. sourcecode:: yaml
+
+  datasource:
+    AliYun:
+      metadata_urls: ["http://100.100.100.200", "http://instance-data:8773"]
+      support_xen: false
```
[the remark this answers is not in the document]
Comment (bittopaz) [reply doc/rtd/topics/datasources/aliyun.rst:27]: Thanks for reviewing this PR. 
hmm, this is strange then, I got the example from their [doc](https://www.alibabacloud.com/help/doc-detail/57803.htm)
Looks like Alicloud maintains its own version of cloud-init inside.

Anyway, I'll remove this line.

**Comment by bittopaz** [comment]
@OddBloke  Thanks for taking time reviewing the changes. I have updated accordingly and rebased the commits.

**Review by OddBloke — APPROVED** [review]
Thanks for the updates!

(As this is a doc change only, we do not require CLA signature, so this is good to land.)

**Pull request merged**